state/oracledatabase/oracledatabaseaccess.go

```diff
@@ -194,16 +194,16 @@ func (o *oracleDatabaseAccess) doSet(ctx context.Context, db querier, req *state
 		result, err = db.ExecContext(ctx, updateStatement, value, binaryYN, etag, req.Key, *req.ETag)
 	}
 	if err != nil {
-		if req.ETag != nil && *req.ETag != "" {
-			return state.NewETagError(state.ETagMismatch, err)
-		}
 		return err
 	}
 	rows, err := result.RowsAffected()
 	if err != nil {
 		return err
 	}
 	if rows != 1 {
+		if req.ETag != nil && *req.ETag != "" {
+			return state.NewETagError(state.ETagMismatch, err)
```

Comment:
Wait. Was this another eTag issue detected?

Reply:
Between this and #2834 I have fixed plenty of etag errors.

In most cases the errors were simply components not reporting etag failures as state.ETagError errors... In other cases it was much worse: https://github.com/dapr/components-contrib/pull/2834#discussion_r1197032949